src/m365/commands/setup.ts

```diff
@@ -25,8 +36,94 @@ interface CommandArgs {
 interface Options extends GlobalOptions {
   interactive?: boolean;
   scripting?: boolean;
+  skipApp?: boolean;
+}
+
+export enum CliUsageMode {
+  Interactively = 'interactively',
+  Scripting = 'scripting'
+}
+
+export enum CliExperience {
+  Beginner = 'beginner',
+  Proficient = 'proficient'
+}
+
+export enum EntraAppConfig {
+  Create = 'create',
+  UseExisting = 'useExisting',
+  Skip = 'skip'
 }
 
+export enum NewEntraAppScopes {
+  Minimal = 'minimal',
+  All = 'all'
+}
+
+export enum HelpMode {
+  Full = 'full',
+  Options = 'options'
+}
+
+const minimalScopes = [
+  'https://graph.microsoft.com/User.Read'
+];
+const allScopes = [
+  'https://graph.windows.net/Directory.AccessAsUser.All',
+  'https://management.azure.com/user_impersonation',
+  'https://admin.services.crm.dynamics.com/user_impersonation',
+  'https://graph.microsoft.com/AppCatalog.ReadWrite.All',
```

Comment:
These scopes belong with the setup command, which is why I decided to keep them here so that they're easily available in the context where we need them.

Reply:
Yes that is true but still this is kinda of a config thing. Keeping it in a separate file will also give us more opportunities to use it in other places. There first thing I may think of is the flow we consider to build which will use dev proxy to find minimal scopes for each command. This flow could also validate if we have those scopes in this list for the app we setup.

Reply:
Nice idea. Will adjust